tests/functional/adapter/grants/test_model_grants.py

```diff
@@ -0,0 +1,305 @@
+    config:
+      materialized: table
+      grants:
+        select: ["{{ env_var('DBT_TEST_USER_1') }}"]
+"""
+
+user2_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select: ["{{ env_var('DBT_TEST_USER_2') }}"]
+"""
+
+multiple_users_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select: ["{{ env_var('DBT_TEST_USER_1') }}", "{{ env_var('DBT_TEST_USER_2') }}"]
+"""
+
+multiple_privileges_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select: ["{{ env_var('DBT_TEST_USER_1') }}"]
+        insert: ["{{ env_var('DBT_TEST_USER_2') }}"]
+"""
+
+# table materialization single select
+extended_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select:
+          user: ["{{ env_var('DBT_TEST_USER_1') }}"]
+          group: ["{{ env_var('DBT_TEST_GROUP_1') }}"]
+          role:  ["{{ env_var('DBT_TEST_ROLE_1') }}"]
+"""
+
+# table materialization change select
+extended2_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select:
+          user: ["{{ env_var('DBT_TEST_USER_2') }}"]
+          group: ["{{ env_var('DBT_TEST_GROUP_2') }}"]
+          role:  ["{{ env_var('DBT_TEST_ROLE_2') }}"]
+"""
+
+# table materialization multiple grantees
+extended_multiple_grantees_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select:
+          user: ["{{ env_var('DBT_TEST_USER_1') }}", "{{ env_var('DBT_TEST_USER_2') }}"]
+          group: ["{{ env_var('DBT_TEST_GROUP_1') }}", "{{ env_var('DBT_TEST_GROUP_2') }}"]
+          role:  ["{{ env_var('DBT_TEST_ROLE_1') }}", "{{ env_var('DBT_TEST_ROLE_2') }}"]
+"""
+# table materialization multiple privileges
+extended_multiple_privileges_table_model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      materialized: table
+      grants:
+        select:
+          user: ["{{ env_var('DBT_TEST_USER_1') }}"]
+          group: ["{{ env_var('DBT_TEST_GROUP_1') }}"]
+          role:  ["{{ env_var('DBT_TEST_ROLE_1') }}"]
+        insert:
+          user: ["{{ env_var('DBT_TEST_USER_2') }}"]
+          group: ["{{ env_var('DBT_TEST_GROUP_2') }}"]
+          role:  ["{{ env_var('DBT_TEST_ROLE_2') }}"]
+"""
+
+
+class TestModelGrantsRedshift(BaseGrantsRedshift):
+    @pytest.fixture(scope="class")
+    def models(self):
+        updated_schema = self.interpolate_name_overrides(model_schema_yml)
+        return {
+            "my_model.sql": my_model_sql,
+            "schema.yml": updated_schema,
+        }
+
+    def test_view_table_grants(self, project, get_test_users, get_test_groups, get_test_roles):
```

Comment:
Running locally I see issues with the expected/actual grants (it looks like the actual grants includes a superset of all users). Kind of hard to debug since there are so many test cases in this test. Can we break this up into separate test cases?

Reply:
Hi Colin, I can certainly do that. 

However, the tests that are failing successfully pass in my local environment. Is there a way to set up my local environment to mirror dbt's local + remote testing environment? I follow the [contributing guide](https://github.com/dbt-labs/dbt-redshift/blob/18c315004d53e50fb822e93fb8eb25cfeb6eebdf/CONTRIBUTING.md), but some test results conflict between local and remote environment.

Reply:
Remotely, it seems the fixtures are not manifesting even with the fix. https://github.com/dbt-labs/dbt-redshift/actions/runs/6461848915/job/17542365369?pr=626#step:7:275
```
get_test_users = ['dbt_test_user_1', 'dbt_test_user_2', 'dbt_test_user_3']
get_test_groups = [], get_test_roles = []
```

Reply:
@soksamnanglim took another look at the tests and it looks like they're being invoked twice in some places. Simplified by making the "base" classes standard test classes since they shouldn't need to be inherited

Reply:
also confirmed that the roles/groups are setup in our test DB via:
`select * from SVV_ROLES;` and `select * from pg_group;`

Reply:
After running `revoke SELECT on "database"."schema"."my_model" from dbt_test_user_1, group dbt_test_group_1, role dbt_test_role_1;`, `test_model_grants` successfully passes for me.

It seems that the privileges were not revoked properly the first time, but seems like they are automatically revoking fine now. 

Can we try manually revoking those privileges from the dbt testing cluster and running the grant test suite multiple times to ensure its behaving deterministically in the dbt environment?